apps/dh/api-dh/source/DataHub.WebApi/GraphQL/Types/BatchType.cs

```diff
@@ -32,7 +34,22 @@ public BatchType(IDataLoaderContextAccessor accessor)
             Field(x => x.ExecutionTimeStart, nullable: true).Description("The execution start time.");
             Field(x => x.ExecutionTimeEnd, nullable: true).Description("The execution end time.");
             Field(x => x.ProcessType).Description("The process type.");
-            Field(x => x.CreatedByUserId).Description("Id of the user who started the batch.");
+
+            Field<NonNullGraphType<StringGraphType>>("CreatedByUserName")
+                .Resolve()
+                .WithScope()
+                .WithService<IMarketParticipantClient>()
+                .Resolve((context, markpart) =>
+                {
+                    var loader = accessor.Context!.GetOrAddLoader("SearchUsersAsync", () => markpart.SearchUsersAsync(
```

Comment:
Is it not possible to search for the id?

Reply:
Idea was to fetch and cache all users and do lookup after, so as to not make a markpart call per batch